pkg/reconciler/pipelinerun/pipelinespec/pipelinespec.go

```diff
@@ -22,32 +22,37 @@ import (
 	"fmt"
 
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tresources "github.com/tektoncd/pipeline/pkg/reconciler/taskrun/resources"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // GetPipeline is a function used to retrieve Pipelines.
-type GetPipeline func(context.Context, string) (v1beta1.PipelineObject, error)
+type GetPipeline func(context.Context, string) (v1beta1.PipelineObject, *v1beta1.ConfigSource, error)
```

Comment:
maybe a terrible idea but PipelineObject is an interface right? Can we add a function to that GetConfigSource() that returns a ConfigSource object instead of changing the function signature?

Reply:
We probably don't want to embed location data in to pipelines themselves, and I'd lean towards not adding extra interface methods that the base pipeline type would need to `return nil` for.

You'd probably want another interface type that embeds this one that would wrap the Pipeline with the provenance data, but that would also modify the func signature so I don't feel too strongly vs leaving it as-is. 🤷

Reply:
Hmm ok...I did say it might be terrible idea 😛 

I think what we have is fine for now, though it'd be nice if we didn't have to pass around the ConfigSource as much as we do now